stdlib,configs,tests: Remove deprecated Resource classes usage

BobbyRBruce opened the pull request:
The following resource classes are deprecated:

* `Resource`: Replaced with `obtain_resource`.
* `CustomResource`: Replaced with specializations.
* `CustomDiskImage`: Replaced with `DiskImageResource`.

There was still some usage in our codebase. In our "configs/example" scripts, in our tests, and some references, in comments, in the main gem5 source. This removes them.

This stops gem5 from regularly throwing warnings as we're using deprecated classes.

BobbyRBruce committed:
stdlib,configs,tests: Remove `Resource` class use

This class is deprecated, but was still used in various example
configuration scriots and tests. This patch replaces it with the
`obtain_resource` function.

Change-Id: I0c89bf17783ccaaafc18072aaeefb5d1e207bc55

BobbyRBruce committed:
configs: Remove `CustomDiskImageResource` use

The class is deprecated but was still used in the SPEC example scripts.
This patch replaces it with the `DiskImageResource` class.

Change-Id: Ie0697fe59a3d737b05eb45ff3bc964f42b0387e0

BobbyRBruce committed:
configs,tests: Remove `CustomResource` use

This class is deprecated but was still used in example scripts and
mentioned, incorrectly, in comments in the pyunit tests. This patch
removes these.

Change-Id: Icb6d02f47a5b72cd58551e5dcd59cc72d6a91a01

powerjg commented on src/python/gem5/resources/workload.py:
```diff
@@ -148,7 +148,8 @@ class Workload(AbstractWorkload):
     # Optionally we can override a parameter in the workload. In this example
     # we are going to run this workload with a difference kernel.
-    workload.set_parameter("kernel", Resource("arm64-linux-kernel-4.14.134"))
+    workload.set_parameter("kernel", \
+        obtain_resource("arm64-linux-kernel-4.14.134"))
```
I don't think the backslash is needed. I'm not sure how this got passed black...

```suggestion
    workload.set_parameter("kernel",
        obtain_resource("arm64-linux-kernel-4.14.134")
    )
```

BobbyRBruce replied:
This code is in a docstring. It's an example of usage. Black probably just ignores it.

Regardless, I'll incorporate this.

powerjg commented on tests/gem5/configs/arm_boot_exit_run.py:
```diff
@@ -199,15 +199,15 @@
 
 # Set the Full System workload.
 board.set_kernel_disk_workload(
```
Does it make sense to pack this up into a workload?

BobbyRBruce replied:
Maybe, but I also like to have some "raw" set_workload examples around as well. Both are helpful.

Either way, I think I can do this in another PR if required?

powerjg commented on tests/gem5/configs/boot_kvm_fork_run.py:
```diff
@@ -179,10 +179,10 @@
 # Set the Full System workload.
 motherboard.set_kernel_disk_workload(
-    kernel=Resource(
+    kernel=obtain_resource(
```
Same thing here... should this be a workload?

BobbyRBruce replied:
Same comment as before. It could be but I think having some set_workload examples is good too. I don't think it's in the scope of this patch to do that conversion.

powerjg reviewed (commented):
A couple of small things.

BobbyRBruce committed:
stdlib: Remove '\' in Workload docstring example

This example shows how to use the Workload. The backslash is not correct Python and would fail if used in this way.

Co-authored-by: Jason Lowe-Power <jason@lowepower.com>
